server/clip_server/executors/clip_hg.py

```diff
@@ -0,0 +1,197 @@
+            If you set ``use_default_preprocessing=True`` when creating this encoder,
+            then the ``tensor`` arrays should have the shape ``[H, W, 3]``, and be in
+            the RGB color format with ``dtype=uint8``.
+            If you set ``use_default_preprocessing=False`` when creating this encoder,
+            then you need to ensure that the images you pass in are already
+            pre-processed. This means that they are all the same size (for batching) -
+            the CLIP model was trained on images of the size ``224 x 224``, and that
+            they are of the shape ``[3, H, W]``  with ``dtype=float32``. They should
+            also be normalized (values between 0 and 1).
+        :param parameters: A dictionary that contains parameters to control encoding.
+            The accepted keys are ``traversal_paths`` and ``batch_size`` - in their
+            absence their corresponding default values are used.
+        """
+        traversal_paths = parameters.get('traversal_paths', self.traversal_paths)
+        batch_size = parameters.get('batch_size', self.batch_size)
+        overwrite_embeddings = parameters.get(
+            'overwrite_embeddings', self.overwrite_embeddings
+        )
+        text_docs = DocumentArray(
+            filter(
+                lambda x: (
+                    bool(x.text) and (overwrite_embeddings or x.embedding is None)
+                ),
+                docs[traversal_paths],
+            )
+        )
+        image_docs = DocumentArray(
+            filter(
+                lambda x: (
+                    (x.tensor is not None or x.blob != b'' or x.uri)
+                    and (overwrite_embeddings or x.embedding is None)
+                ),
+                docs[traversal_paths],
+            )
+        )
+
+        with torch.inference_mode():
+            for batch in text_docs.batch(batch_size=batch_size):
+                self._encode_texts(batch)
+
+            for batch in image_docs.batch(batch_size=batch_size):
+                self._encode_images(batch)
+
+    def _encode_images(self, batch: DocumentArray):
+        """Encode images using the CLIP image encoder."""
+        tensors_batch = []
+        for d in batch:
+            if d.blob != b'':
+                d_tmp = deepcopy(d)
+                d_tmp.convert_blob_to_image_tensor()
+                tensor = d_tmp.tensor
+            elif d.tensor is not None:
+                tensor = d.tensor
+            else:
+                # must be uri
+                d_tmp = deepcopy(d)
+                d_tmp.load_uri_to_image_tensor()
+                tensor = d_tmp.tensor
+            tensors_batch.append(tensor)
+        if self.use_default_preprocessing:
+            tensor = self._preprocess_images(tensors_batch)
+        else:
+            tensor = {
+                'pixel_values': torch.tensor(
+                    batch.tensors, dtype=torch.float32, device=self.device
+                )
+            }
+
```

Comment:
separate `preprocess` and `encode` steps into different sub-functions so as to get better monitoring, refer to this https://github.com/jina-ai/clip-as-service/pull/738